ARCHITECTURE.md

```diff
@@ -0,0 +1,134 @@
+
+### The Pipeline and Callbacks
+Upon being passed to the `notify` function, the report object will also be populated with information provided by a series of callbacks created with the `client` object.  Registered with the `registerCallback` method each callback will be provided the `report` in turn and can populate it with additional information if necessary.
+
+The pipeline object itself is responsible for executing these callbacks as a series of closures until they have all been able to access the `report` object and modify its content.  The pipeline object is merely a method of utilising the callbacks, and does not need to be modified per framework.
+
+The default callbacks, registered through the `client` objects `registerDefaultCallbacks` use the `Resolver` and its `Request` objects to extract data about the current environment in the server, where the error-causing request originated from, and any more meta-data it can report.
+
+There are two callbacks registered with the pipeline automatically by the client: `BreadcrumbData` which ensures that the recorded `Breadcrumb` objects are attached to the `report`, and `NotificationSkipper` which stops the notification process in the event that the notification should not be sent i.e. when missing an `api_key` or a non-releasing `releaseStage`.
+
+It is recommended to create callback functions that can extract additional information from the framework to use attach as metadata if necessary.  These callbacks are automatically wrapped in a `CallbackBridge` when registered to the pipeline which ensures they will automatically be called.
+
+### The HTTPClient
+Once the `report` object has been populated by the `pipeline` a callback is finally triggered in the `notify` function that will send the `report` to the HTTPClient.  This object exists on the `client` and is intiated with a `guzzle` client that it will use to call off to the configured endpoint.
+
+This object will queue each `report` object it is given until `send` is called.  At this point it will iteratively create a single object containing all of the data from the `queue`, and ensure that the payload is correctly set up to be sent while remaining under the payload size limit.
+
+Once the payload has been fully constructed it will be posted to the configured endpoint via the guzzle object.
+
+The HTTPClient is also responsible for the `deploy` call.
+
+### Breadcrumbs
+Bugsnag tracks a series of actions manually dictated by the user to be sent to the Bugsnag notify endpoint along with an exception.  These actions are stored as breadcrumb objects, and are stored in the recorder object in the `client`.  The recorder acts like a circular array, storing up to 25 breadcrumb objects at a time before the oldest breadcrumbs get overwritten.  This limit is imposed to ensure the size of the payload sent to Bugsnag does not exceed the API limits
```

Comment:
On other notifiers some breadcrumbs are also captured automatically, is that the case with PHP?